File: app/src/main/java/me/arndc/example/patterns/mvp/java/MainContract.java
package me.arndc.example.patterns.mvp.java;

import me.arndc.example.patterns.mvp.java.domain.Pattern;

interface MainContract {

    interface View {

        void showInformation(final Pattern pattern);

        void showError(final String message);
    }

    interface UserActionListener {

        void requestInformation(final String abbreviation);
    }
}

File: app/src/main/java/me/arndc/example/patterns/mvp/kotlin/MainContract.kt
package me.arndc.example.patterns.mvp.kotlin

import me.arndc.example.patterns.mvp.kotlin.domain.Pattern

interface MainContract {

    interface View {

        fun showInformation(pattern: Pattern)

        fun showError(message: String)
    }

    interface UserActionListener {

        fun requestInformation(abbreviation: String)
    }
}
